DOCS-9841-DOCS-9528-@Lazy()-duke (#1770)
mulesoft/docs-mule-runtime-3: modules/ROOT/pages/dataweave-variables.adoc

endif::[]
:keywords: studio, anypoint, esb, transform, transformer, format, aggregate, rename, split, filter convert, xml, json, csv, pojo, java object, metadata, dataweave, data weave, datamapper, dwl, dfl, dw, output structure, input structure, map, mapping

> DataWeave is a functional programming language in which variables behave just like functions. DataWeave eagerly evaluates variables, the same way it evaluates arguments to functions. In addition, DataWeave variables are immutable. 

Reviewer: eagerly?

Reviewer: I wonder if we could have one sentence instead of three: that in DW, variables not only behave and are evaluated in the same way as functions but also are immutable.

Author: That sounds better. However, our team is likely to add more about immutability, so I want to keep that sentence separate. It's a stop-gap at the moment.

---
addSuffix("hello", toUpper, " world") //result is "HELLO WORLD"
----

== @Lazy() Annotation

> An internal annotation, `@Lazy()`, can appear in autocompletion suggestions when you are writing a DataWeave script in Studio. This annotation is not intended for general use and does not improve the performance of your scripts. 

Reviewer: Just an alternative:
```suggestion
When you are writing a DataWeave script in Studio, you might see the internal annotation `@Lazy()` in autocompletion suggestions. This annotation is not intended for general use and does not improve the performance of your scripts. 
```
Or even more simply?
```suggestion
If you see the internal annotation `@Lazy()` in autocompletion suggestions when you are writing a DataWeave script in Studio, ignore it.
```
